Task 9

DmitryVDV opened the pull request.

DmitryVDV committed:
Task - 9 done some tasks

1. Добавить регистрацию
2. Логинить пользователя сразу после регистрации
3. Добавить возможность +добавлять, редактировать и просматривать
только своикарточки

Бонус задачи
1. Можно зарегистрироваться в один клик при помощи одной из соц. сетей
(Facebook/Vkontakte/Twitter).
2. Для хранения omniauth токена и секретного ключа используются
переменные окружения (ENV)

DmitryVDV committed:
Task-9 done email an password edition

4. Добавить редактирование своего аккаунта (почту, пароль)

DmitryVDV committed:
Task-9 add custom devise views

DmitryVDV committed:
task-9 add test

X1ting commented on app/assets/stylesheets/users.scss:
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Users controller here.
```
зачем оставляешь пустые файлы?

DmitryVDV replied:
уберу

X1ting commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,45 @@
+require 'awesome_print'
+require 'pry'
```
это реально нужно?

X1ting replied:
ну вот эти все три строки

DmitryVDV replied:
уберу

X1ting commented on spec/factories/users.rb:
```diff
@@ -1,8 +1,9 @@
 FactoryGirl.define do
   factory :user do
-    email "my@mail.ru"
+    sequence(:email) { |n| "someone#{rand(1..100000000)}@mail.com" }
```
у тебя же есть n переменная, которая отвечает за уникальность, зачем еще rand дергать

DmitryVDV replied:
n всегда с 1 начинается, а база у меня не очищается тестовая, на уникальности вылетает тест. Поэтому рандомно сделал.

X1ting replied:
а database_cleaner не пробовал?

X1ting commented on spec/factories/users.rb:
```diff
@@ -1,8 +1,9 @@
     password "MyString"
-
+    provider "vk"
+    sequence(:uid) { |n| rand(1000000..10000000) }
```
тут тоже самое

X1ting commented on spec/interactor/card_update_reviewdate.rb:
```diff
@@ -0,0 +1,18 @@
+
+      fill_in "translation", with: @card.translated_text
+      click_button "Проверить"
+      expect(page).to have_content("Перевод верный", wait:15)
```
wait:15 тощно без пробела должно писаться?

DmitryVDV replied:
исправлю

X1ting commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,45 @@
+    params.require(:user).permit(:email)
+  end
+
+  def user_params_password
```
лучша наверное будет выглядеть как password_params и email_params сверху

DmitryVDV replied:
поправлю

DmitryVDV committed:
Task-9 - finished test

1. Add Databasecleaner for test
2. Add test for password update
3. Fixed mentor task

X1ting commented on app/controllers/cards_controller.rb:
```diff
@@ -14,10 +16,14 @@ def new
+
+    #@card = Card.create(card_params)
+
+    if user.cards.create!(card_params)
```
а зачем create! а не create

X1ting commented on app/controllers/cards_controller.rb:
```diff
@@ -1,8 +1,8 @@
 
   def index
-    @cards = Card.all
+    @cards = Card.where(user_id: current_user.id)
```
why not `current_user.cards` ?

X1ting commented on app/controllers/cards_controller.rb:
```diff
@@ -14,10 +14,13 @@ def new
-      redirect_to action: :index
+    puts card_params
+
+    user = User.find(current_user.id)
```
what reason for it? why you not use `current_user`

X1ting commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,45 @@
+  end
+
+  def update_email
+    @user = User.find(current_user.id)
```
why not `@user = current_user` ?

X1ting commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,45 @@
+  end
+
+  def update_password
+    @user = User.find(current_user.id)
```
why not @user = current_user ?
and if you do it multiple times, better move to before_action

X1ting commented on app/interactors/show_user_card.rb:
```diff
@@ -0,0 +1,14 @@
+
+  def call
+    # Note that the first_or_create method automatically sets the provider and
+    # uid fields when creating a new user. The first_or_create! method operates
```
its usefull comment?

X1ting commented on app/models/card.rb:
```diff
@@ -16,7 +16,7 @@ class Card < ApplicationRecord
   # using (strip) and set all symbols to downcase
   def data_strip_downcase
-    # Unicode must edit via  mb_chars ! 
+    # Unicode must edit via  mb_chars !
```
useless comment

X1ting commented on app/views/devise/registrations/new.html.erb:
```diff
@@ -0,0 +1,18 @@
+<div class="container">
+<h2>Регистрация</h2>
```
malformed indentation in this file

X1ting commented on app/views/devise/sessions/new.html.erb:
```diff
@@ -0,0 +1,15 @@
+<div class="container">
+<h2>Вход</h2>
+<%= simple_form_for(resource, as: resource_name, url: session_path(resource_name)) do |f| %>
```
indentations

X1ting commented on app/views/users/update_password.html.erb:
```diff
@@ -0,0 +1,2 @@
+<h1>Users#update_password</h1>
```
useless file

X1ting commented on spec/views/users/edit.html.erb_spec.rb:
```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+require 'database_cleaner'
+
+DatabaseCleaner.strategy = :truncation, {:only => %w[users cards]}
```
you can move DatabaseCleaner setup to rails_helper, and forgot about it)

X1ting commented on spec/views/users/edit.html.erb_spec.rb:
```diff
@@ -0,0 +1,37 @@
+
+DatabaseCleaner.strategy = :truncation, {:only => %w[users cards]}
+
+RSpec.describe 'users/edit.html.erb', type: :feature do
```
just `describe`
Why `type: :feature` in spec/views ?

X1ting commented on app/views/users/_userform.html.erb:
```diff
@@ -0,0 +1,21 @@
+<div class="container">
```
indentation
and always files like this, called _form.html.erb

X1ting commented on app/controllers/cards_controller.rb:
```diff
@@ -14,10 +14,13 @@ def new
-    @card = Card.new(card_params)
-    if @card.save
-      redirect_to action: :index
+    puts card_params
```
`puts` ?

DmitryVDV committed:
Task-9 changes after review

DmitryVDV committed:
Task-9 changes after review
